File: solutions/length-of-last-word.py
# Length of Last Word
# Split string by spaces. Iterate from back to front, checking for a
# nonempty string. If iterates past length of string, return 0
# Time: O(n) | Space: O(n)

from test import test

def lengthOfLastWord(s: str) -> int:
    l = s.split(' ')
    i = 1
    while i <= len(l) and not(l[-i]):
        i += 1
    return len(l[-i]) if i <= len(l) else 0

# Driver Code
cases = [(' ', 0), ('a     ', 1), ('', 0)]
test(lengthOfLastWord, cases)

File: solutions/lengthOfLastWord.js
// Length of Last Word 

const test = require('./test');

/**
 * Split string by space, return length of last word lol
 * 
 * @param {string} s
 * @return {number}
 */
const lengthOfLastWord = s => {
    let words = s.split(' ');
    // Iterate through words from back to front until non-empty word found
    let i = words.length - 1;
    while (i > 0 && words[i].length == 0) {
        i--;
    }
    return words[i].length;
};

// Second method
const lengthOfLastWord2 = s => {
    let words = s.trim().split(' ');
    return words[words.length - 1].length;
};


// Driver Code
const cases = [
    ["Hello World", 5],
    ["", 0],
    ["a", 1],
    ["    ", 0],
    ["a ", 1],
    ["a    ", 1],
    [" a", 1],
    ["     a", 1]
];
test(cases, lengthOfLastWord);
test(cases, lengthOfLastWord2);
